src/Router/types.ts

```diff
@@ -1,53 +1,53 @@
 import { RouteConfig } from "found"
-import { ComponentType } from "react"
 import { Environment } from "relay-runtime"
+import { Container, SubscribeProps } from "unstated"
 import { Breakpoint } from "Utils/Responsive"
 import { ContextProps } from "../Components/Artsy"
 
-type ReactComponent = ComponentType<any>
 type HistoryProtocol = "browser" | "hash" | "memory"
 
 export interface AppConfig {
   historyProtocol?: HistoryProtocol
   initialBreakpoint?: Breakpoint
   initialRoute?: string
+  initialState?: Array<Container<any>>
+  initialAppState?: object // TODO: Deprecated
   routes: RouteConfig
-  url?: string
   user?: User
+
+  subscribe?: {
+    to: SubscribeProps["to"]
+    onChange: (...instances: Array<Container<any>>) => void
+  }
+
+  // TODO: What is this used for?
+  url?: string
 }
 
+export type App = React.ComponentType<{}>
+
 export interface ClientResolveProps {
-  ClientApp: ReactComponent
+  ClientApp: App
 }
 
 export interface ServerResolveProps {
-  ServerApp?: ReactComponent
+  ServerApp?: App
   redirect?: string
   status?: string
 }
 
-export interface AppShellProps {
-  loadableState?: {
-    getScriptTag: () => string
-  }
-  data?: Array<object>
-}
```

Comment:
Colocating typings with the one place they’re being used increases readability and refactor-ability of the codebase. A ‘types’ module is I think mostly helpful when e.g. typing external data that comes into the system and various parts of the system need to use those typings.